### PR TITLE
Enhanced _group_by Function and cancel statement

### PR DESCRIPTION
**Pull request by hellozepp**
release note:
- Support for transpiling grouping set、cube、rollup clause
- Support for transpiling hash functions to murmurhash3_32
- kill job syntax to cancel job syntax

### CODE REVIEW

File: sqlglot/dialects/clickzetta.py [hunk 1]
```diff
@@ -84,6 +85,8 @@ def _anonymous_func(self: ClickZetta.Generator, expression: exp.Anonymous) -> st
         return f"'Asia/Shanghai'"
     elif expression.this.upper() == 'GROUPING':
         return f"GROUPING_ID({self.expressions(expression, flat=True)})"
+    elif expression.this.upper() == 'HASH':
```
Comment (czrobert) [inline sqlglot/dialects/clickzetta.py]: Proof needed here. Does hash and murmurhash3_32 behave same?

File: sqlglot/dialects/clickzetta.py [hunk 2]
```diff
@@ -160,6 +163,58 @@ def adjust_day_of_week(self: ClickZetta.Generator, expression: exp.DayOfWeek):
     else:
         return f'DAYOFWEEK({self.sql(expression.this)})'
 
+
+def _group_sql(self: ClickZetta.Generator, expression: exp.Group):
```
Comment (czrobert) [inline sqlglot/dialects/clickzetta.py]: Maybe transform.preprocess is a better and more light weight way here, no need to re-implement whole group_sql func.

File: sqlglot/dialects/clickzetta.py [hunk 3]
```diff
@@ -242,6 +297,8 @@ class Generator(Spark.Generator):
             exp.DateAdd: date_add_sql,
             exp.RegexpExtract: regexp_extract_sql,
             exp.DayOfWeek: adjust_day_of_week,
+            exp.Group: _group_sql,
+            exp.Kill: lambda self, e: f"CANCEL JOB {self.sql(e.this)}",
```
Comment (czrobert) [inline sqlglot/dialects/clickzetta.py]: Not very necessary to transpile a cancel job here. Our job id is auto generated, even a exp.Kill is transpiled, it has no change to run correctly.

**Review by czrobert — CHANGES_REQUESTED** [review]

File: sqlglot/dialects/clickzetta.py (hunk 1 above)
Comment (hellozepp) [reply sqlglot/dialects/clickzetta.py]: Currently we do not support hive hash, remove the code, add starrocks murmur_hash3_32 conversion

File: sqlglot/dialects/clickzetta.py [hunk 4]
```diff
@@ -84,6 +85,8 @@ def _anonymous_func(self: ClickZetta.Generator, expression: exp.Anonymous) -> st
         return f"'Asia/Shanghai'"
     elif expression.this.upper() == 'GROUPING':
         return f"GROUPING_ID({self.expressions(expression, flat=True)})"
+    elif expression.this.upper() == 'MURMUR_HASH3_32':
+        return f"murmurhash3_32({self.sql(expression.expressions[0])})"
```
Comment (czrobert) [inline sqlglot/dialects/clickzetta.py]: by default, all transpiled keywords are upper case

**Pull request merged**